src/controllers/bulk/bulkcontroller.h

```diff
@@ -46,51 +46,52 @@ class BulkController : public Controller {
                    struct libusb_device_descriptor *desc);
     virtual ~BulkController();
 
-    virtual QString presetExtension();
+    virtual QString presetExtension() override;
 
-    virtual ControllerPresetPointer getPreset() const {
+    virtual ControllerPresetPointer getPreset() const override {
         HidControllerPreset* pClone = new HidControllerPreset();
         *pClone = m_preset;
         return ControllerPresetPointer(pClone);
     }
 
-    virtual bool savePreset(const QString fileName) const;
+    virtual bool savePreset(const QString fileName) const override;
```

Comment:
Use either 'virtual' for the first declaration or 'override' in derived classes, but not both.